x-pack/plugins/enterprise_search/public/applications/shared/indexing_status/indexing_status_logic.test.ts

```diff
@@ -0,0 +1,110 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { resetContext } from 'kea';
+
+jest.mock('../http', () => ({
+  HttpLogic: {
+    values: { http: { get: jest.fn() } },
+  },
+}));
+import { HttpLogic } from '../http';
+
+jest.mock('../flash_messages', () => ({
+  flashAPIErrors: jest.fn(),
+}));
+import { flashAPIErrors } from '../flash_messages';
+
+import { IndexingStatusLogic } from './indexing_status_logic';
+
+describe('IndexingStatusLogic', () => {
+  jest.useFakeTimers();
+
+  let unmount: any;
+
+  const mockStatusResponse = {
+    percentageComplete: 50,
+    numDocumentsWithErrors: 3,
+    activeReindexJobId: 1,
+  };
+
+  beforeEach(() => {
+    jest.clearAllMocks();
+    resetContext({});
+    unmount = IndexingStatusLogic.mount();
+  });
+
+  it('has expected default values', () => {
+    expect(IndexingStatusLogic.values).toEqual({
+      percentageComplete: 100,
+      numDocumentsWithErrors: 0,
+    });
+  });
+
+  describe('setIndexingStatus', () => {
+    it('sets reducers', () => {
+      IndexingStatusLogic.actions.setIndexingStatus(mockStatusResponse);
+
+      expect(IndexingStatusLogic.values.percentageComplete).toEqual(
+        mockStatusResponse.percentageComplete
+      );
+      expect(IndexingStatusLogic.values.numDocumentsWithErrors).toEqual(
+        mockStatusResponse.numDocumentsWithErrors
+      );
+    });
+  });
+
+  describe('fetchIndexingStatus', () => {
+    const statusPath = '/api/workplace_search/path/123';
+    const onComplete = jest.fn();
+    const TIMEOUT = 3000;
+
+    it('calls API and sets values', async () => {
+      const setIndexingStatusSpy = jest.spyOn(IndexingStatusLogic.actions, 'setIndexingStatus');
+      const promise = Promise.resolve(mockStatusResponse);
+      (HttpLogic.values.http.get as jest.Mock).mockReturnValue(promise);
+
+      IndexingStatusLogic.actions.fetchIndexingStatus({ statusPath, onComplete });
+      jest.advanceTimersByTime(TIMEOUT);
+
+      expect(HttpLogic.values.http.get).toHaveBeenCalledWith(statusPath);
+      await promise;
+
+      expect(setIndexingStatusSpy).toHaveBeenCalledWith(mockStatusResponse);
+    });
+
+    it('handles error', async () => {
+      const promise = Promise.reject('An error occured');
+      (HttpLogic.values.http.get as jest.Mock).mockReturnValue(promise);
+
+      IndexingStatusLogic.actions.fetchIndexingStatus({ statusPath, onComplete });
+      jest.advanceTimersByTime(TIMEOUT);
+
+      try {
+        await promise;
+      } catch {
+        expect(flashAPIErrors).toHaveBeenCalledWith('An error occured');
+      }
```

Comment:
@JasonStoltz actually [pointed out to me recently](https://github.com/elastic/kibana/pull/81926#discussion_r514348295) that putting an `expect()` within a `catch()` can lead to false negatives (as it did within that particular PR). It's safer to do this:

```suggestion
      try {
        await promise;
      } catch {
        // Do nothing
      }
      expect(flashAPIErrors).toHaveBeenCalledWith('An error occured');
```

That being said I think there's a couple existing logic files that I haven't updated to use this pattern, definitely a tech debt to pay down